src/Bicep.Core/Syntax/Converters/StatementSyntaxConverter.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Bicep.Core.Parsing;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Core.Syntax.Converters
+{
+    public static class StatementSyntaxConverter
```

Comment:
The source code of the converters is basically copied from the decompiler with some tweaks. We cannot use the decompiler directly here because it won't work when a JSON template contains a nested template. Also, I wanted to avoid having `Core` reference `Decompiler`.